src/metadata.js

```diff
@@ -55,6 +55,38 @@ export function getObjects(projectId, objectUris) {
 }
 
 /**
+ * Loads all objects by query (fetches all pages, one by one)
+ *
+ * @method getObjectsByQuery
+ * @param {String} projectId id of the project
+ * @param {Object} options (see developer.gooddata.com/api endpoint: /gdc/md/{project_id}/objects/query)
+ *        - category {String} for example 'dataSets' or 'projectDashboard'
+ *        - mode {String} 'enriched' or 'raw'
+ *        - author {String} the URI of the author of the metadata objects
+ *        - limit {number} default is 50 (also maximum)
+ * @return {Promise<Array>} array of returned objects
+ */
+export function getObjectsByQuery(projectId, options) {
+    let items = [];
+
+    function getOnePage(uri) {
+        return get(uri)
+            .then((r) => {
+                const currentItems = _get(r, 'objects.items');
+                items = [...items, ...currentItems];
+
+                const nextUri = _get(r, 'objects.paging.next');
+                return nextUri ? getOnePage(nextUri) : items;
+            });
+    }
+
+    const uri = `/gdc/md/${projectId}/objects/query`;
+    const query = pick({ limit: 50, ...options }, ['category', 'mode', 'author', 'limit']);
+    return getOnePage(uri + queryString(query));
+}
```

Comment:
I would like advices on making the code more elegant...
 ie. maybe what about this `items.push(..._get(r, 'objects.items'));` instead of L75-76?